packages/store/addon/-private/system/core-store.ts

```diff
@@ -328,10 +328,12 @@ abstract class CoreStore extends Service {
           number: 'NumberTransform',
           string: 'StringTransform',
         };
+        type Mappings = keyof typeof Mapping;
+        const keys = Object.keys(Mapping) as Mappings[];
         let shouldWarn = false;
 
         let owner = getOwner(this);
-        Object.keys(Mapping).forEach((attributeType: keyof typeof Mapping) => {
+        keys.forEach((attributeType: Mappings) => {
```

Comment:
The cast `as Mappings[]` should result in this type inferring correctly, so if you like you can drop the explicit annotation here:

```suggestion
        keys.forEach((attributeType) => {
```

Reply:
This can be removed but I additionally cleaned up the naming so done as a commit vs accepting suggestion.